### DISCUSSION TITLE
Monthly gem5 Developers' Meeting - September 12th at 4pm (UTC)

### DISCUSSION THREAD

**Discussion by ivanaamit** [gem5/gem5#1503] [Announcements]
The monthly gem5 Developers' Meeting will be held on September 12th at 4pm UTC.

### Agenda:
- #1498 
- prebuilt boards: What are they meant for? #1478  
- #1486 
- #1545 

### Meeting Archive:
- [Video](https://youtu.be/lWuo1bdhEiM)
- [Slides](https://github.com/user-attachments/files/16983921/2024-09-12.monthly.meeting.pptx)

**Comment by erin-le** [comment]
Notes from today's meeting:

# 9-12-2024 Dev meeting
- participants
    - erin, jason, ivana, mahyar, harshil, giacomo, jonathan haley, hemant bhandari, daniel carvalho, richard cooper, rit


- issues and prs are growing

- reason for dev meetings: communicate with community about what dev team is doing

- agenda: deprecating things, adding demo board, Mahyar's findings

## Demo boards

- stdlib: reproducible, accurate
- eventually: want "known good configurations" - some % of accuracy against actual hardware, difficult
- demo boards - establish infrastructure of accuracy testing. Not configurable/extensible, want them to be fixed.
- want to show that the workloads/suites run

### Goal

- (showed screenshot from gem5 resources website)
- want a new tab on the website that shows what the status is on the corresponding demo board
- time, simtime, instr, other stats
- Other cool ideas
    - discussion 754
    - dashboard for currently running wklds
    - see what has/hasn't been tested, see if the results are reasonable

- Giacomo has questions
    - 1. run daily tests on board?
        a. Run only on stable - once per release cycle
    - 2. How to update? - prev answer answered this
    - 3. Adding fixed model: consider microarch is tightly coupled with architecture
        - divergence/convergence to real hardware when change
    - 4. depends on kernel config - maybe provide config file
        - extract kernel from disk img
        - how to extract kernel from arm ubuntu? already know how to do x86
    - Jason's response to 3 and 4: need to enumerate the caveats to the data
    - Giacomo: make it explicit that we will not match real hardware; don't overpromise

- Jason wants to work with ARM people to figure out configurations
    - "configurations" covers everything
    - especially kernel configs, arm platform configs, release config, hardware
    - Giacomo suggests: have piece of hardware, enable all supported extensions, model based on that
        - they can't share microarch
    - Jason: wants outside reviewers to increase community truse

## Deprecating configs/common

- encourage users to write python code, not bash scripts
- presenting every possible parameter is bad for research
    - options.py is misleading, people will continue using it if not deprecated
- blockers: arm fs tests, gpu scripts

- Gia: Arm fs tests don't need command line params
    - some arm files in common are useful, move them? Not tightly coupled with options.py
    
- Jason: update arm fs tests to not have params, move cores directory out of common, leave starter fs and starter se alone for now, convert to stdlib in long term

- Giacomo tried to combine starter fs and stdlib, couldn't do it
    - he suspects that similar issues will occur with starter se
    - Jason thinks it would be possible to do starter se

- GPU scripts difficult to port, discussion skipped since GPU people weren't present

- Giacomo: grep file for options, extract the ones used from options.py, put in a separate config script to decouple from options.py

## change_id

- pr 1486
- will be merged today

## Mahyar

- Arm paired memory ops
- Working to develop known good config for Arm Ampere Altra

- paired memory instrs cause discrepancy

- short samples of execution of NPB workloads, run on both gem5 and hardware

- samples:
    - gem5 is from 1.6 times faster to 6.6 times slower
    - gem5 accurate for data cache access
    - 1x to 4x in store instr

    - error in num cycles correlated to num data cache accesses
    - error in data cache accesses is mostly explained by error in num load and store

- giacomo comment: gem5 microop vs macroop?

- translate paired mem instrs to one microop instead of possibly multiple, results in hypothetical graph

- Conclusion
    - error in data cache accesses is correlated and explainable by error in num load and store instrs
    - we think the root cause is how PairMemOps are translated to microops in gem5

- Send Giacomo slides, binaries, things needed to test
